tests/python_tests/samples/conftest.py

```diff
@@ -0,0 +1,141 @@
+import subprocess # nosec B404
+import os
+import tempfile
+import pytest
+import shutil
+import logging
+import gc
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Define model names and directories
+MODELS = {
+    "TinyLlama-1.1B-Chat-v1.0": { 
+        "name": "TinyLlama/TinyLlama-1.1B-Chat-v1.0",
+        "convert_args": []
+    },
+    "TinyLlama-1.1B-intermediate-step-1431k-3T": {
+        "name": "TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T",
+        "convert_args": ['--trust-remote-code']
+    },  
+    "WhisperTiny": {
+        "name": "openai/whisper-tiny",
+        "convert_args": ['--trust-remote-code']
+    },
+    "open_llama_3b_v2": {
+        "name": "openlm-research/open_llama_3b_v2",
+        "convert_args": ['--trust-remote-code', '--weight-format', 'fp16']
+    },
+    "Qwen-7B-Chat": {
+        "name": "Qwen/Qwen-7B-Chat",
+        "convert_args": ['--trust-remote-code', '--weight-format', 'fp16']
+    },
+    "Qwen1.5-7B-Chat": {
+        "name": "Qwen/Qwen1.5-7B-Chat",
+        "convert_args": ['--trust-remote-code', '--weight-format', 'fp16']
+    },
+    "phi-2": {
+        "name": "microsoft/phi-2",
+        "convert_args": ['--trust-remote-code', '--weight-format', 'fp16']
+    },
+    "notus-7b-v1": {
+        "name": "argilla/notus-7b-v1",
+        "convert_args": ['--trust-remote-code', '--weight-format', 'fp16']
+    }
+}
+
+TEST_FILES = {
+    "how_are_you_doing_today.wav": "https://storage.openvinotoolkit.org/models_contrib/speech/2021.2/librispeech_s5/how_are_you_doing_today.wav",
+    "adapter_model.safetensors": "https://huggingface.co/smangrul/tinyllama_lora_sql/resolve/main/adapter_model.safetensors"
+}
+
+TEMP_DIR = os.environ.get("TEMP_DIR", tempfile.mkdtemp())
+MODELS_DIR = os.path.join(TEMP_DIR, "test_models")
+TEST_DATA = os.path.join(TEMP_DIR, "test_data")
+
+SAMPLES_PY_DIR = os.environ.get("SAMPLES_PY_DIR", os.getcwd())
+SAMPLES_CPP_DIR = os.environ.get("SAMPLES_CPP_DIR", os.getcwd())
+
+# A shared fixture to hold data
+@pytest.fixture(scope="session")
+def shared_data():
+    return {}
```

Comment:
I would avoid sharing some data between tests to avoid extra complexity.

The same for models - let's assume that samples are running on unique models to have a better coverage

Reply:
removed